src/views/forms/TradeForm/MarketLeverageInput.tsx

```diff
@@ -56,9 +56,9 @@ export const MarketLeverageInput = ({
     postOrderSize,
   });
 
-  const maxLeverage = initialMarginFraction
-    ? BIG_NUMBERS.ONE.div(initialMarginFraction)
-    : MustBigNumber(10);
+  const preferredIMF = effectiveInitialMarginFraction ?? initialMarginFraction;
+
+  const maxLeverage = preferredIMF ? BIG_NUMBERS.ONE.div(preferredIMF) : MustBigNumber(10);
 
   const leverageOptions = maxLeverage.lt(10) ? [1, 2, 3, 4, 5] : [1, 2, 3, 5, 10];
```

Comment:
We may need to adjust these options here accordingly. e.g the larges option should not be greater than `maxLeverage`. Or we can show the options and disable the buttons that are bigger than maxLeverage

Reply:
ah my bad i was under the assumption leverage state setting would be capped by max leverage and that would be a good enough experience.

this is the experience i was imagining for this
- user clicks x10
- if max leverage is x3.03, slider is set to x3.03 which is the max.
- link: https://www.loom.com/share/8ff4ba24f6e8439ca1f98dc433e29853?sid=cdb21f53-6cbf-4f2b-a899-e8716318fffe